actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -128,9 +128,19 @@ def initialize(set:, ast:, controller:, default_action:, to:, formatted:, via:,
           @internal           = options.delete(:internal)
           @scope_options      = scope_params[:options]
 
-          path_params = ast.find_all(&:symbol?).map(&:to_sym)
+          path_params = []
+          wildcard_options = {}
+          ast.each do |node|
+            if node.symbol?
+              path_params << node.to_sym
+            elsif node.star? && formatted != false
```

Comment:
It feels like goal of `formatted` here is to check that it is `truthy`. Literally it's checking that it isn't `false`, but is it necessary to explicitly look for `false`?

Would it work to do something like this?

```suggestion
            elsif node.star? && !formatted
```
Or would the possibility of it being `nil` be an issue? It could also be a choice here to be very explicit

Reply:
Yeah, in this case we need to check specifically for `false`. Formatted will often be `nil` and in that case we still want to run this code. If would only be `false` if that option was explicitly passed (e.g. `get '*pages', to: 'pages#show', format: false`).